dvc/stage.py

```diff
@@ -566,6 +569,9 @@ def load(repo, fname):
         Stage._check_isfile(repo, fname)
 
         d = load_stage_file_fobj(repo.tree.open(fname), fname)
+        # Making a deepcopy since the original structure
```

Comment:
any specific reasons for this, @efiop ? It's a good practice to avoid side effects like this. Is it easy to change this guys? I would also add something like `d = freeze(load_stage_file_fobj(repo.tree.open(fname), fname))` if python has something like freeze.

Reply:
Python have immutable collections like tuples and frozendicts, and no freeze() function like that, that could be implemented though. The bigger change would be rewriting all deps and outs loading functions to not pop from given structures.